leap_ec/multiobjective/nsga2.py

```diff
@@ -28,13 +29,18 @@ def nsga_2(max_generations: int, pop_size: int, problem, representation,
           "A Fast and Elitist Multiobjective Genetic Algorithm: NSGA-II." IEEE
           transactions on evolutionary computation 6, no. 2 (2002): 182-197.
 
```

Comment:
Let's add a sentence explaining what we mean by "generalized."  Something like:

```python
"""
NSGA-II multi-objective evolutionary algorithm.  This classic algorithm relies on the idea of "non-dominated sorting" and de-crowding to evolve a diverse Pareto front.  The "generalized" NSGA-II we implement here differs slightly from the canonical algorithm, in that we default to a faster sorting algorithm devised by Burlacu (2022).

- Deb, Kalyanmoy, Amrit Pratap, Sameer Agarwal, and T. A. M. T. Meyarivan.
          "A Fast and Elitist Multiobjective Genetic Algorithm: NSGA-II." IEEE
          transactions on evolutionary computation 6, no. 2 (2002): 182-197.

 - Bogdan Burlacu. 2022. Rank-based Non-dominated Sorting. arXiv.
          DOI:https://doi.org/10.48550/ARXIV.2203.13654 

If you wish the algorithm to use the original NSGA-II behavior instead (which runs much slower), you can select the original operator by passing in `rank_fun=fast_nondominated_sort`.
"""
```

Reply:
I added your comment into the docstring and formatted it to match the rest. I didn't have anything else to add to it.